app/views/unattended/provisioning_templates/snippet/kickstart_rhsm.erb

```diff
@@ -0,0 +1,17 @@
+<%#
+kind: snippet
+name: kickstart_rhsm
+model: ProvisioningTemplate
+snippet: true
+description: |
+  Generate Anaconda RHSM configuration for RHEL registration
```

Comment:
Technically registration isn't limited to RHEL. You can register all systems that support the `rhsm` macro, for example CentOS 9 to your Katello instance. However, I am missing that critical piece, which EL 9 supports via `--server-hostname`. https://pykickstart.readthedocs.io/en/latest/kickstart-docs.html#rhsm has the complete reference.

Oddly enough that documentation says it's EL8+ while https://github.com/theforeman/foreman/pull/9352/files#discussion_r945636919 says the register part is EL9+.